app/redux/wallet/actions.ts

```diff
@@ -99,6 +103,25 @@ export const unlockWallet = ({ password }: { password: string }) => async (dispa
   const requestApiSelection = isWalletOnly && !meta.remoteApi;
   return { success: true, forceNetworkSelection: hasNetworks && (!isNetworkExist || requestApiSelection), isWalletOnly };
 };
+export const unlockCurrentWallet = (password: string) => async (dispatch: AppThDispatch, getState: GetState) => {
+  const {
+    wallet: { currentWalletPath },
+  } = getState();
+  if (!currentWalletPath) {
+    dispatch(setUiError(new Error(`Can not find wallet in ${currentWalletPath}`)));
+    return { success: false };
+  }
+  return dispatch(unlockWallet(currentWalletPath, password));
+};
+
+export const closeWallet = () => (dispatch: AppThDispatch) => {
+  dispatch(setCurrentWalletPath(null));
```

Comment:
maybe for this better to use tools like this - https://react-redux.js.org/api/batch

 better to use object or so, but if we cannot, batch working well

Reply:
Yep, I made a [proposal for this case](https://github.com/spacemeshos/smapp/issues/799#p2) some time ago. We should refactor such a mess for sure. Built-in `batch` may work as well.
But I'd prefer to make it within another PR, and since we have no tests, it may take some extra time for testing all the changes 🤔

Also, I think that some chunks with such a pattern may be removed in case we start refactoring redux reducers with `redux-toolkit`.

Reply:
agree